addons/voice/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java

```diff
@@ -0,0 +1,215 @@
+		if ((null == text) || text.isEmpty()) {
+			throw new TTSException("The passed text is null or empty");
+		}
+		if (!this.voices.contains(voice)) {
+			throw new TTSException("The passed voice is unsupported");
+		}
+		boolean isAudioFormatSupported = false;
+		for (AudioFormat currentAudioFormat : this.audioFormats) {
+			if (currentAudioFormat.isCompatible(requestedFormat)) {
+				isAudioFormatSupported = true;
+				break;
+			}
+		}
+		if (!isAudioFormatSupported) {
+			throw new TTSException("The passed AudioFormat is unsupported");
+		}
+
+		// now create the input stream for given text, locale, format. There is
+		// only a default voice
+		try {
+			File cacheAudioFile = voiceRssImpl.getTextToSpeechAsFile(this.apiKey, text,
+					voice.getLocale().toLanguageTag(), requestedFormat.getCodec());
+			if (cacheAudioFile == null) {
+				throw new TTSException("Could not read from VoiceRSS service");
+			}
+			AudioStream audioStream = new VoiceRSSAudioStream(cacheAudioFile, requestedFormat);
+			return audioStream;
+		} catch (IOException ex) {
+			throw new TTSException("Could not read from VoiceRSS service: " + ex.getMessage(), ex);
+		}
+	}
+
+	/**
+	 * Initializes this.voices.
+	 *
+	 * @return The voices of this instance
+	 */
+	private final HashSet<Voice> initVoices() {
+		HashSet<Voice> voices = new HashSet<Voice>();
+		Set<Locale> locales = voiceRssImpl.getAvailableLocales();
+		for (Locale local : locales) {
+			Set<String> voiceLabels = voiceRssImpl.getAvailableVoices(local);
+			for (String voiceLabel : voiceLabels) {
+				voices.add(new VoiceRSSVoice(local, voiceLabel));
+			}
+		}
+		return voices;
+	}
+
+	/**
+	 * Initializes this.audioFormats
+	 *
+	 * @return The audio formats of this instance
+	 */
+	private final HashSet<AudioFormat> initAudioFormats() {
+		HashSet<AudioFormat> audioFormats = new HashSet<AudioFormat>();
+		Set<String> formats = voiceRssImpl.getAvailableAudioFormats();
+		for (String format : formats) {
+			audioFormats.add(getAudioFormat(format));
+		}
+		return audioFormats;
+	}
+
+	/**
+	 * Up to now only MP3 supported.
+	 */
+	private final AudioFormat getAudioFormat(String format) {
+		// MP3 format
+		if (AudioFormat.CODEC_MP3.equals(format)) {
+			// we use by default: MP3, 44khz_16bit_mono
+			Boolean bigEndian = null; // not used here
+			Integer bitDepth = 16;
+			Integer bitRate = null;
+			Long frequency = 44000L;
+
+			return new AudioFormat(AudioFormat.CONTAINER_NONE, AudioFormat.CODEC_MP3, bigEndian, bitDepth, bitRate,
+					frequency);
+		} else {
+			throw new RuntimeException("Audio format " + format + "not yet supported");
+		}
+	}
+
+	private final CachedVoiceRSSCloudImplementation initVoiceImplementation() {
+		CachedVoiceRSSCloudImplementation apiImpl = new CachedVoiceRSSCloudImplementation(getCacheFolderName());
+		return apiImpl;
+	}
+
+	String getCacheFolderName() {
+		String folderName = ConfigConstants.getUserDataFolder();
+		if (!folderName.endsWith(File.separator)) {
```

Comment:
Agreed. So I will always add a trailing slash assuming that the "future" PR will always remove trailing slash.